temporalio/converter.py

```diff
@@ -427,3 +440,38 @@ async def decode_payloads(
     if not payloads or not payloads.payloads:
         return []
     return await converter.decode(payloads.payloads)
+
+
+def _type_hints_from_func(
+    func: Callable[..., Any], eval_str: bool
+) -> Tuple[Optional[List[Type]], Optional[Type]]:
+    """Extracts the type hints from the function.
+
+    Args:
+        func: Function to extract hints from.
+        eval_str: Whether to use ``eval_str`` (only supported on Python >= 3.10)
+
+    Returns:
+        Tuple containing parameter types and return type. The parameter types
+        will be None if there are any non-positional parameters or if any of the
+        parameters to not have an annotation that represents a class.
+    """
+    sig_kwargs: Mapping[str, Any] = {}
+    # Only supported on >= 3.10
+    if sys.version_info >= (3, 10):
+        sig_kwargs = {"eval_str": eval_str}
+    sig = inspect.signature(func, **sig_kwargs)
```

Comment:
It can. I didn't personally see having different paths call the same method w/ different args as clearer than different paths building the args and then one single invocation point.

Reply:
Updated to suggestion